src/pyk/kastManip.py

```diff
@@ -437,6 +437,23 @@ def _labels_to_dots(k: KInner) -> KInner:
     return bottom_up(_labels_to_dots, kast)
 
 
+def extract_cells(kast: KInner, cells: Collection[str]) -> KInner:
+    contains_selected_cell: Dict[str, bool] = {}
+
+    def _extract_cells(k: KInner) -> KInner:
+        if type(k) is KApply and k.is_cell:
+            contains_selected_cell[k.label.name] = (k.label.name in cells) or any(
+                type(arg) is KApply and arg.is_cell and contains_selected_cell[arg.label.name] for arg in k.args
+            )
+
+            if not contains_selected_cell[k.label.name]:
+                return DOTS
+
+        return k
```

Comment:
Is this a valid simplification?

```suggestion
    def _extract_cells(k: KInner) -> KInner:
        if (
            type(k) is KApply and 
            k.is_cell and
            not k.label.name in cells and
            all(arg == DOTS for arg in k.args)
        ):
            return DOTS
        return k
```